pkg/ccl/streamingccl/streamclient/client.go

```diff
@@ -147,6 +137,39 @@ func NewStreamClient(streamAddress streamingccl.StreamAddress) (Client, error) {
 	return streamClient, nil
 }
 
+// NewClientFromTopology iterates through each stream address in the topology
+// and returns the first client it's able to successfully connect to.
+func NewClientFromTopology(ctx context.Context, topology *jobspb.StreamTopology) (Client, error) {
+	streamAddresses := make(map[string]struct{})
+	for _, partition := range topology.PartitionInfo {
+		streamAddresses[partition.Address] = struct{}{}
+	}
+
+	var combinedError error = nil
+	for address := range streamAddresses {
+		streamAddress := streamingccl.StreamAddress(address)
+		client, err := NewStreamClient(streamAddress)
+		if err == nil {
+			dialErr := client.Dial(ctx)
+
+			if dialErr == nil {
+				return client, err
+			}
+
+			err = errors.CombineErrors(err, dialErr)
```

Comment:
I can't even imagine how hard it would be to read a combined error over 5 or 10 nodes;
Anyway, I think log.Error is in order for each attempt.